File: Daily Challenge/C/Substring - First Occurrence 18.7.2025.c
#include <stdio.h>
#include <string.h>

int main() {
    char s1[1001], s2[101];
    scanf("%s %s", s1, s2);
    int min = 1000, max = -1;
    for (int i = 0; s2[i]; i++) {
        for (int j = 0; s1[j]; j++) {
            if (s2[i] == s1[j]) {
                if (j < min) min = j;
                if (j > max) max = j;
                break;
            }
        }
    }
    for (int i = min; i <= max; i++)
        printf("%c", s1[i]);
    return 0;
}

File: Daily Challenge/Java/Substring - First Occurrence 18.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s1 = sc.next(), s2 = sc.next();
        int min = s1.length(), max = -1;
        for (char c : s2.toCharArray()) {
            int idx = s1.indexOf(c);
            if (idx < min) min = idx;
            if (idx > max) max = idx;
        }
        System.out.print(s1.substring(min, max + 1));
    }
}